### ISSUE TITLE
the logPx log_var is zero

### ISSUE THREAD

**Issue by benlaitang** [JeremyCCHsu/vae-npvc#11]
i am curious anout https://github.com/JeremyCCHsu/vae-npvc/blob/master/model/vae.py#L124
            logPx = tf.reduce_mean(
                GaussianLogDensity(
                    slim.flatten(x),
                    slim.flatten(xh),
                    tf.zeros_like(slim.flatten(xh))),
if log_var is constant 0, leading to loss  GaussianLogDensity is equivalent to MSE？

**Comment by JeremyCCHsu** [comment]
Yes, it's intended. Constant variance prevents overfitting.

**Comment by benlaitang** [comment]
> Yes, it's intended. Constant variance prevents overfitting.

I noticed that D_KL loss https://github.com/JeremyCCHsu/vae-npvc/blob/master/model/vae.py#L113 
 and logPx https://github.com/JeremyCCHsu/vae-npvc/blob/master/model/vae.py#L120 is related ~N(0,1) where u is 0 and log var is 1, this is not same to paper. it is just a trick for training?
Actually, I finished the VAE code for training multiple speaker, but it did not  work, I use the KL loss+ MSE loss.


![image](https://user-images.githubusercontent.com/22116685/64308697-bf396b00-cfcc-11e9-9ac1-6ba956e03f0e.png)

**Comment by benlaitang** [comment]
> Yes, it's intended. Constant variance prevents overfitting.



> Yes, it's intended. Constant variance prevents overfitting.

**Comment by JeremyCCHsu** [comment]
I don't quite understand your question... Var=0 --> log var = 0. Actually, tuning log var could help in some cases.

update: typo: var=1 --> log var =1

**Comment by benlaitang** [comment]
> I don't quite understand your question... Var=0 --> log var = 0. Actually, tuning log var could help in some cases.

log var =0 is var = 1. 
Can you help check the VAE loss, give me some advice? Now it did not work.
